### PR TITLE
Add HP-UX network facts

### PR DESCRIPTION
**Pull request by pdelared**
Added support for HPUX network facts

### CODE REVIEW

File: lib/ansible/module_utils/facts.py [hunk 1]
```diff
@@ -2048,6 +2048,57 @@ def merge_default_interface(self, defaults, interfaces, ip_type):
...
+            return self.facts
+        self.get_default_interfaces()
+        interfaces = self.get_interfaces_info()
+        self.facts['interfaces'] = interfaces.keys()
+        for iface in interfaces:
+                self.facts[iface] = interfaces[iface]
+        return self.facts
+
+    def get_default_interfaces(self):
+        rc, out, err = module.run_command("/usr/bin/netstat -nr", use_unsafe_shell=True)
```
Comment (abadger) [inline lib/ansible/module_utils/facts.py:2075]: What's the reason for using the shell here?  There's no redirection, pipes, or other needs for the shell that I can see.

File: lib/ansible/module_utils/facts.py [hunk 2]
```diff
@@ -2048,6 +2048,57 @@ def merge_default_interface(self, defaults, interfaces, ip_type):
...
+        for line in lines:
+                words = line.split()
+                if len(words) > 1:
+                    if words[0] == 'default':
+                        self.facts['default_interface'] = words[4]
+                        self.facts['default_gateway'] = words[1]
+
+    def get_interfaces_info(self):
+        interfaces = {}
+        rc, out, err = module.run_command("/usr/bin/netstat -ni", use_unsafe_shell=True)
```
Comment (abadger) [inline lib/ansible/module_utils/facts.py:2086]: Same question here about unsafe_shell as in get_default_interfaces()

**Comment by abadger** [comment]
I don't have any HP-UX machines so I'll have to trust that the code works :-)  I just have a few questions about the idioms in you're using in the code.

File: lib/ansible/module_utils/facts.py (hunk 1 above)
Comment (pdelared) [inline lib/ansible/module_utils/facts.py:2075]: I based my code on class HP UX(Hardware). I'm not a python expert. Feel free to correct my bad idioms an i will do the test on HP UX box

File: lib/ansible/module_utils/facts.py (hunk 2 above)
Comment (pdelared) [inline lib/ansible/module_utils/facts.py:2086]: same as above

**Comment by amenonsen** [comment]
AFAICS, the needs_info request has been answered, and the PR is ready to merge with use_unsafe_shell=False in two places and the spelling correction noted above.

**Pull request merged**